### PR TITLE
Clean up JdbcLockRegistry, ZookeeperLockRegistry cache automatically

### LINKED ISSUE #3672: Clean up JdbcLockRegistry, ZookeeperLockRegistry cache automatically
**Expected Behavior**
If you're okay
I'm working on it and I'm trying to post a PR.

**Current Behavior**
now
If you use a lot of different keys
Currently, you have to call `expireUnusedOlderThan()` manually.

after change
Developers can use it without manually invoking it.
If you do not want local caching, you can also disable caching by setting `capacity`.

**Context**

Similar to an issue before, I'd like to contribute.
https://github.com/spring-projects/spring-integration/issues/3655

### PR DESCRIPTION
**Pull request by Meteorkor**
Resolves https://github.com/spring-projects/spring-integration/issues/3672

Currently, if you don't manually call expireUnusedOlderThan(), OutOfMemory can happen.
Therefore, I changed the code so that if the cache exceeds a setting capacity, it is automatically deleted.

### `JdbcLockRegistry`
default capacity : `100_000`, about 21MB
image

### `ZookeeperLockRegistry`
if default of capacity is `100_000`, about 83MB
image

if default of capacity is `30_000`, about 25MB
image

because `ZookeeperLockRegistry` default of capacity  is `30_000`

### CODE REVIEW

**Comment by artembilan** [comment]
> Merge branch 'main' of 

We prefer do not `merge`, but `rebase`. Less noisy commits in the PR.
Plus easier for us to squash then before merging back to `main`.

See more info here: https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc.

Also: I prefer do not see `Agreed`, `Done`, `Fix`, `OK` etc. for every single my review when you are just going to push the fix for my suggestions.
I will see the change and your agreement with me in your next commit, so what is the point to double post for that ? 😄 
On the other hand, I fully encourage to continue discussion if you have something in mind and my position is somehow wrong or not professional since I might not know something well in the subject we are having in the PR.

Just letting you to know what is a collaboration over here with me 😄 

Thank you for you contribution again!

**Review by artembilan — CHANGES_REQUESTED** [review]
Just a couple concerns.
I can fix them on merge if you prefer.

Thanks

File: spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java
Comment (Meteorkor) [reply spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java:137]: I think that synchronization is also necessary here.

I think there are two reasons

at first
At the time of get()
If you `Map.put()` on another thread, the order of the nodes stored inside the `LinkedHashMap` is changed and somtimes an error may occur(`ConcurrentModificationException`).

the second
Since the `accessOrder` of the current `LinkedHashMap` is set to `true`, if the size of the Map has reached the threshold
When `LinkedHashMap.get()` is called, the delete action will occur.

File: spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java
Comment (Meteorkor) [reply spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java:102]: I misunderstood.
Oh, you mentioned changing the field name from `capacity` to `cacheCapacity`.

I will change

File: spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java
Comment (artembilan) [reply spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java:137]: OK. Convinced. Thanks.

So, fix `cacheCapacity` property, please, and I'll merge this.

**Review by artembilan — APPROVED** [review]
LGTM.

Will merge after green GH Action on this PR.

**Comment by artembilan** [comment]
Thanks again for your contribution!

**Pull request merged**
